core/src/main/java/org/infinispan/interceptors/impl/EntryWrappingInterceptor.java

```diff
@@ -365,34 +400,29 @@ private void removeFromContextOnRetry(InvocationContext ctx, Collection<?> keys)
 
    @Override
    public final Object visitRemoveCommand(InvocationContext ctx, RemoveCommand command) throws Throwable {
-      wrapEntryIfNeeded(ctx, command);
-      return setSkipRemoteGetsAndInvokeNextForDataCommand(ctx, command);
+      return setSkipRemoteGetsAndInvokeNextForDataCommand(ctx, command, wrapEntryIfNeeded(ctx, command));
    }
 
    @Override
    public Object visitRemoveExpiredCommand(InvocationContext ctx, RemoveExpiredCommand command) throws Throwable {
-      entryFactory.wrapEntryForExpired(ctx, command.getKey(), command.getSegment(),
-            ignoreOwnership(command) || canRead(command));
-      return setSkipRemoteGetsAndInvokeNextForDataCommand(ctx, command);
+      entryFactory.wrapEntryForExpired(ctx, command.getKey(), command.getSegment());
```

Comment:
I would have made `wrapEntryForExpired` return a stage, just so `visitRemoveExpiredCommand` looks like all the others :)